examples/frontend/pages/cc/[project]/explore/index.tsx

```diff
@@ -0,0 +1,179 @@
+// @ts-nocheck
+/** @jsxImportSource theme-ui */
+
+import { useState, useEffect } from 'react'
+import { useSetRecoilState } from 'recoil'
+import { useRouter } from 'next/router'
+import { utils } from 'near-api-js'
+import { Button, Box, Link, Spinner } from 'theme-ui'
+
+import { useNFTMethod, useNFTContract } from '@cura/hooks'
+import { MediaObject } from '@cura/components'
+
+import Layout from '../../../../containers/Layout'
+import { getFrameWidth } from 'utils/frame-width'
+import { mapPathToProject } from 'utils/path-to-project'
+import { alertMessageState } from '../../../../state/recoil'
+
+const CONTRACT_RANDOM_GAS = utils.format.parseNearAmount('0.00000000020') // 200 Tgas
+
+const NFT_PER_PAGE = 4
+
+const Explore = () => {
+    const router = useRouter()
+    const setAlertMessage = useSetRecoilState(alertMessageState)
+
+    const contractAdress = router && mapPathToProject(router.asPath)
+    const project = `cc/${router.query.project}`
+
+    // get total supply, which we need to detect if all NFTs are loaded
+    const { data: totalSupply } = useNFTMethod(
+        contractAdress,
+        'nft_total_supply',
+        {}
+    )
+
+    // hook that contains all the logic of explore page
+    const { items, loading, nextPage } = useNFTExplore(
+        contractAdress,
+        NFT_PER_PAGE
+    )
+
+    // load next page
+    function loadMore() {
+        // if all NFTs are loaded or we are already loading more then do nothing
+        if (totalSupply <= items.length || loading) {
+            return
+        }
+        try {
+            nextPage()
+        } catch (error) {
+            console.error(error)
+            setAlertMessage(error.toString())
+        }
+    }
+
+    // If user scroll to bottom, show next page
+    const handleScroll = () => {
```

Comment:
Could have used something like this? https://github.com/ankeetmaini/react-infinite-scroll-component
Makes life a bit easier. :)